scripts/prod.js

```diff
@@ -14,20 +14,28 @@ const prod = async (isDeploy, noDeps) => {
     await generateGraphQLArtifacts()
   } catch (e) {
     console.log('ERR generating artifacts', e)
+    process.exit(1)
   }
 
   console.log('starting webpack build')
   try {
-    await Promise.all([
+    const webpacks = await Promise.all([
       runChild(
         `yarn webpack --config ./scripts/webpack/prod.servers.config.js --no-stats --env=noDeps=${noDeps}`
       ),
       runChild(
         `yarn webpack --config ./scripts/webpack/prod.client.config.js --no-stats --env=minimize=${isDeploy}`
       )
     ])
+    webpacks.forEach(
+      function (exitCode) {
+        if (exitCode > 0)
+          throw new Error(`Error during one of the webpack build processes. Webpacks error codes ${webpacks}`)
+      }
+    )
```

Comment:
-1 I think we should fix `runChild` instead to reject the promise if the process did not successfully exit. Something along the lines of
```ts
const runChild = (cmd) => {
  return new Promise((resolve, reject) => {
    const build = cp.exec(cmd).on('exit', (code, signal) => {
      if (code) {
        reject(new Error(`Received non-zero exit code ${code}`))
      }
      if (signal) {
        reject(new Error(`Received signal ${signal}`))
      }
      resolve()
    })
    build.stderr.pipe(process.stderr)
  })
}
```

Reply:
this feels right! only nit is to use an `else if` so we don't resolve after we reject